examples/hosted-login-code-flow/src/main/resources/application.yml

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright 2017 Okta, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+logging:
+  level:
+    root: INFO
+    org.springframework.web: DEBUG
+    org.springframework.security: DEBUG
+
+okta:
+  oauth2:
+    # default is unset
+    customLoginRoute: /login
+
+    # default is '/login'
+    redirectUri: /login/okta
```

Comment:
Why do the defaults need to be changed? Does it work when these aren't specified, or are these settings specific to this example?

Reply:
Actually, I want to pick your brain about this.  I'm not super happy with the way this works at the moment. (this is also related to your next comment).
I'm trying to keep the the _hosted_ page stuff optional, and default to a standard redirect (takes less config, easier to debug, etc).

This project still redirects, it just redirects to the oauth2 filter at: `/login/okta` which grabs the `state` and then renders the widget template `login.html`

Longer term it would be _nice_ to handle this all in the same request, and just forward to the `login.html` template (to save a hop, and configuration)